src/utils/poolData/kyber.ts

```diff
@@ -1,41 +1,63 @@
 import { BigNumber, Contract } from 'ethers'
 
 import {
+  ChainId,
   KYBER_FACTORY,
   KYBER_FACTORY_ABI,
+  KYBER_FACTORY_POLYGON,
   KYBER_POOL_ABI,
   TEN_POW_18,
 } from 'utils/constants/constants'
-import { WETH } from 'utils/constants/tokens'
+
 import { getProvider } from 'utils/provider'
+import { getWETH } from 'utils/weth'
 
 type KyberBalances = {
   tokenBalance: BigNumber
   wethBalance: BigNumber
 }
 
+function getFactoryAddress(chainId: ChainId) {
+  switch (chainId) {
+    case ChainId.ethereum:
+      return KYBER_FACTORY
+    case ChainId.polygon:
+      return KYBER_FACTORY_POLYGON
+  }
+}
+
 export async function getKyberLiquidity(
-  tokenAddress: string
+  tokenAddress: string,
+  chainId: ChainId
 ): Promise<KyberBalances> {
   let response: KyberBalances = {
     tokenBalance: BigNumber.from(0),
     wethBalance: BigNumber.from(0),
   }
-  const provider = getProvider()
-  const factoryInstance = await new Contract(
-    KYBER_FACTORY,
-    KYBER_FACTORY_ABI,
-    provider
-  )
 
-  const pools = await factoryInstance.getPools(tokenAddress, WETH)
-  if (!pools[0]) return response
+  try {
+    const provider = getProvider(chainId)
+    const WETH = getWETH(chainId)
+    const factoryAddress = getFactoryAddress(chainId)
+    const factoryInstance = await new Contract(
+      factoryAddress,
+      KYBER_FACTORY_ABI,
+      provider
+    )
+
+    const pools = await factoryInstance.getPools(tokenAddress, WETH)
+    if (!pools[0]) return response
 
-  const pairContract = await new Contract(pools[0], KYBER_POOL_ABI, provider)
-  const [tokenBalance, wethBalance] = await pairContract.getReserves()
+    const pairContract = await new Contract(pools[0], KYBER_POOL_ABI, provider)
+    const [tokenBalance, wethBalance] = await pairContract.getReserves()
 
-  response.tokenBalance = tokenBalance.div(TEN_POW_18)
-  response.wethBalance = tokenBalance.div(wethBalance)
+    response.tokenBalance = tokenBalance.div(TEN_POW_18)
+    response.wethBalance = tokenBalance.div(wethBalance)
 
-  return response
+    return response
+  } catch (error) {
+    console.log('Error getting liquidity from kyber')
+    console.log(error)
```

Comment:
Sometimes this function does throw errors:

```
Error: cannot estimate gas; transaction may fail or may require manual gas limit (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\": \"2.0\", \"id\": 44, \"error\": {\"code\": -32000, \"message\": \"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"to\":\"0x546c79662e028b661dfb4767664d0273184e4dd1\",\"data\":\"0x5b1dc86f000000000000000000000000d6df932a45c0f255f85145f286ea0b292b21c90b0000000000000000000000007ceb23fd6bc0add59e62ac25578270cff1b9f619\"},\"latest\"],\"id\":44,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://polygon-mainnet.g.alchemy.com/v2/L9jbLT6_-U9rEuvLoMXVvUYoZQx04pEI"}, method="call", transaction={"to":"0x546C79662E028B661dFB4767664d0273184E4dD1","data":"0x5b1dc86f000000000000000000000000d6df932a45c0f255f85145f286ea0b292b21c90b0000000000000000000000007ceb23fd6bc0add59e62ac25578270cff1b9f619"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.0.17)
    at Logger.makeError (index.ts:225)
    at Logger.throwError (index.ts:237)
    at checkError (json-rpc-provider.ts:68)
    at JsonRpcProvider.<anonymous> (json-rpc-provider.ts:463)
    at Generator.throw (<anonymous>)
    at rejected (json-rpc-provider.ts:2)
```
don't know if this would be due to low liquidity as well?

Reply:
In general, another problem might be that on Polygon Kyber doesn't have any pool with `WETH` as far as I see it. Looking at the tokens list [here](https://dmm.exchange/#/pools/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2), it might not even be supported. Should we switch to ETH here?

Reply:
Re: WETH vs ETH
As I understand it, all the liquidity in dexs is usually a WETH swap underneath, they just do the wrapping/unwrapping for you. Specifically on Polygon, there's no ETH, only WETH. 

On either chain, is this a factor for Kyber? Worth looking into

Reply:
Should be fixed now. Had a wrong address of the Kyber factory on polygon as well.